### ISSUE TITLE
Webcam doesn't work: Uncaught (in promise) NotReadableError: Could not start video source

### ISSUE THREAD

**Issue by sglbl** [gradio-app/gradio#10049]
### Describe the bug

While camera was working on my computer with Gradio 3, gradio 4 and 5 versions somehow doesn't support my camera.
This code works on other computers. (Also the examples about camera on gradio website didn't work on gradio 4 and 5)
- I thought it's about using local http so I used share=True but still the same error with https
- My camera app works fine other than gradio app
- (My camera app wasn't open while I was using gradio app)
- I checked the drivers it seems okay (screenshot2)
- I tried with Opera and Edge. 

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr

def process_img(img):
    return 255 - img

def main():
    demo = gr.Interface(
        fn=process_img,
        inputs=gr.Image(),
        outputs='image',
    )

    demo.launch(show_error=True, debug=True, share=True)

if __name__ == "__main__":
    main()
```

### Screenshot

![image](https://github.com/user-attachments/assets/51e5b081-93fb-4fe2-a6be-e82dbf224b87)
![image](https://github.com/user-attachments/assets/7d7a4dd9-9787-42c3-92bf-02f9e9884aa6)

### Logs

```shell
Traceback (most recent call last):
  File "/home/sglbl/deduce/digit_classification/.venv/lib/python3.10/site-packages/gradio/queueing.py", line 624, in process_events
    response = await route_utils.call_process_api(
  File "/home/sglbl/deduce/digit_classification/.venv/lib/python3.10/site-packages/gradio/route_utils.py", line 323, in call_process_api
    output = await app.get_blocks().process_api(
  File "/home/sglbl/deduce/digit_classification/.venv/lib/python3.10/site-packages/gradio/blocks.py", line 2015, in process_api
    result = await self.call_function(
  File "/home/sglbl/deduce/digit_classification/.venv/lib/python3.10/site-packages/gradio/blocks.py", line 1562, in call_function
    prediction = await anyio.to_thread.run_sync(  # type: ignore
  File "/home/sglbl/deduce/digit_classification/.venv/lib/python3.10/site-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "/home/sglbl/deduce/digit_classification/.venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 2441, in run_sync_in_worker_thread
    return await future
  File "/home/sglbl/deduce/digit_classification/.venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 943, in run
    result = context.run(func, *args)
  File "/home/sglbl/deduce/digit_classification/.venv/lib/python3.10/site-packages/gradio/utils.py", line 865, in wrapper
    response = f(*args, **kwargs)
  File "/home/sglbl/deduce/digit_classification/dene.py", line 4, in process_img
    return 255 - img
TypeError: unsupported operand type(s) for -: 'int' and 'NoneType'
(since the camera didn't work, as expected)
```

### System Info

```shell
(digit_classification) sglbl@DEDUCE-PC:~/deduce/digit_classification$ uv pip list | grep gradio
gradio                    5.6.0
gradio-client             1.4.3
```

### Severity

I can work around it

**Comment by abidlabs** [comment]
That's very strange -- I can't repro locally. Is it possible that something else could be using your webcam?

**Comment by sglbl** [comment]
> That's very strange -- I can't repro locally. Is it possible that something else could be using your webcam?

No. When my webcam is open there is a light, every second blinks. It only blinks when my camera is on, and it doesn't blink with gradio 5. [Unless I'm not hacked which I'm not sure how I will understand that]
Also, I'm using WSL2 [my friends who use wsl2 can access to camera with this code so I'm not sure if it's related. Anyway it works with gradio 3 so if you want I can close the issue until somebody else has the same issue and re-opens it.

**Comment by sglbl** [comment]
> That's very strange -- I can't repro locally. Is it possible that something else could be using your webcam?

I solved the issue. I used this website to check my camera [https://webcamtests.com/](url) and it showed me that I have a second camera `"OBS Virtual Camera"`. But in fact I didn't install it with OBS. Even so, I tried to uninstall the virtual camera with win-dshow batch file
 [https://obsproject.com/forum/threads/how-to-uninstall-obs-virtual-camera.146837/](url) but it didn't work. When I completely uninstalled OBS and restarted it worked. 
 I'm not sure why obs (that I explicitly installed without virtual camera) caused this issue. Also because of this error, I never could switch to my integrated camera.
